tests/dualtor/test_grpc_server_failure.py

```diff
@@ -0,0 +1,131 @@
+from tests.common.dualtor.dual_tor_common import ActiveActivePortID
+from tests.common.dualtor.dual_tor_utils import upper_tor_host                              # lgtm[py/unused-import]
+from tests.common.dualtor.dual_tor_utils import lower_tor_host                              # lgtm[py/unused-import]
+from tests.common.dualtor.icmp_responder_control import pause_icmp_responder
+from tests.common.dualtor.nic_simulator_control import ForwardingState
+from tests.common.dualtor.nic_simulator_control import mux_status_from_nic_simulator        # lgtm[py/unused-import]
+from tests.common.dualtor.nic_simulator_control import toggle_active_active_simulator_ports # lgtm[py/unused-import]
+from tests.common.fixtures.ptfhost_utils import run_icmp_responder                          # lgtm[py/unused-import]
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+
+pytestmark = [
+    pytest.mark.topology("dualtor")
+]
+TEST_COUNT = 4
+
+
+@pytest.fixture(scope="module")
+def test_duthost(upper_tor_host, lower_tor_host):
+    today = datetime.datetime.fromtimestamp(time.time())
+    if today.day % 2 == 1:
+        duthost = upper_tor_host
+        portid = ActiveActivePortID.UPPER_TOR
+    else:
+        duthost = lower_tor_host
+        portid = ActiveActivePortID.LOWER_TOR
+    return duthost, portid
+
+
+@pytest.fixture(scope="module")
+def test_mux_ports(active_active_ports):
+    if not active_active_ports:
+        pytest.skip("Skip as no 'active-active' mux ports available")
+    test_ports = random.sample(active_active_ports, min(len(active_active_ports), TEST_COUNT))
+    logging.info("Select ports %s to test", test_ports)
+    return random.sample(active_active_ports, TEST_COUNT)
+
+
+@pytest.fixture(params=[ForwardingState.ACTIVE, ForwardingState.STANDBY])
+def init_port_state(request):
+    return request.param
+
+
+@pytest.fixture
+def setup_test_ports(init_port_state, mux_status_from_nic_simulator, pause_icmp_responder, test_mux_ports):
+
+    def check_forwarding_state(mux_ports, upper_tor_forwarding_state, lower_tor_forwarding_state):
+        mux_status = mux_status_from_nic_simulator(mux_ports)
+        logging.debug(
+            "Check forwarding state of mux ports %s, upper ToR state: %s, lower ToR state: %s",
+            mux_ports,
+            upper_tor_forwarding_state,
+            lower_tor_forwarding_state
+        )
+        logging.debug("Mux status from nic_simulator:\n%s", mux_status)
+        for port in mux_status:
+            if ((mux_status[port][ActiveActivePortID.UPPER_TOR] != upper_tor_forwarding_state) or
+                (mux_status[port][ActiveActivePortID.LOWER_TOR] != lower_tor_forwarding_state)):
+                logging.debug("Port %s mux status is not expected", port)
+                return False
+        return True
+
+    if init_port_state == ForwardingState.STANDBY:
+        pause_icmp_responder(test_mux_ports)
```

Comment:
So the init `standby` state is achieved by pausing icmp responder? 

Just being curious, why don't we `config mux mode` CLI as we already have similar fixture in `nic_simulator_control`? Shouldn't that be easier?

Reply:
This testcase is to verify SONiC could recover from nic failure in `auto` mode. The one you mentioned should be a different scenario?